classifier/classifier.py

```diff
@@ -1,138 +1,247 @@
+
+        self.parser.add_argument("-v", "--version", action='store_true',
+                                 help="show version, filename and exit")
+
+        self.parser.add_argument("-et", "--edittypes", action='store_true',
+                                 help="Edit the list of types and formats")
+
+        self.parser.add_argument("-t", "--types", action='store_true',
+                                 help="Show the current list of types and formats")
+
+        self.parser.add_argument("-r", "--recursive", action='store_true',
+                                 help="Show the current list of types and formats")
+
+        self.parser.add_argument("-st", "--specific-types", type=str, nargs='+',
+                                 help="Move all file extensions, given in the args list, " +
+                                      "in the current directory into the Specific Folder")
+
+        self.parser.add_argument("-sf", "--specific-folder", type=str,
+                                 help="Folder to move Specific File Type")
+
+        self.parser.add_argument("-o", "--output", type=str,
+                                 help="Main directory to put organized folders")
+
+        self.parser.add_argument("-d", "--directory", type=str,
+                                 help="The directory whose files to classify")
+
+        self.parser.add_argument("-dt", "--date", action='store_true',
+                                 help="Organize files by creation date")
+
+        self.parser.add_argument("-df", "--dateformat", type=str,
+                                 help="set the date format using YYYY, MM or DD")
+
+        self.args = self.parser.parse_args()
+        self.dateformat = 'YYYY-MM-DD'
+        self.formats = {}
+        self.dirconf = None
+        self.checkconfig()
+        self.main()
+
+    def checkconfig(self):
+        """ create a default config if not available """
+        if not os.path.isdir(os.path.dirname(CONFIG)):
+            os.makedirs(os.path.dirname(CONFIG))
+        if not os.path.isfile(CONFIG):
+            conffile = open(CONFIG, "w")
+            conffile.write("IGNORE:.part,.desktop\n" +
+                           "Music:.mp3,.aac,.flac,.ogg,.wma,.m4a,.aiff,.wav,.amr\n" +
+                           "Videos:.flv,.ogv,.avi,.mp4,.mpg,.mpeg,.3gp,.mkv,.ts,.webm,.vob,.wmv\n" +
+                           "Pictures:.png,.jpeg,.gif,.jpg,.bmp,.svg,.webp,.psd,.tiff\n" +
+                           "Archives:.rar,.zip,.7z,.gz,.bz2,.tar,.dmg,.tgz,.xz,.iso,.cpio\n" +
+                           "Documents:.txt,.pdf,.doc,.docx,.odf,.xls,.xlsv,.xlsx," +
+                           ".ppt,.pptx,.ppsx,.odp,.odt,.ods,.md,.json,.csv\n" +
+                           "Books:.mobi,.epub,.chm\n" +
+                           "DEBPackages:.deb\n" +
+                           "Programs:.exe,.msi\n" +
+                           "RPMPackages:.rpm")
+            conffile.close()
+        for items in open(CONFIG, "r"):
```

Comment:
It would be better to open files safely using `with`.
Also, reading the file includes `\n` in the formats. It could be avoided:
``` python
with open(CONFIG, 'r') as file:
            for items in file:
                (key, val) = items.replace('\n', '').split(':')
                self.formats[key] = val
```